lambda/src/server/slack.cljs

```diff
@@ -21,62 +26,133 @@
                               (j/get evt :body)])]))
 
 (def base-uri "https://slack.com/api/")
-(def bot-token (-> config :slack :bot-user-oauth-token))
+
+(def scopes ["channels:read"
+             "chat:write"
+             "commands"
+             "groups:read"
+             "users:read"
+             "users:read.email"])
 
 ;; TODO refactir some of content-type & headers into a `def`
 
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;;; Direct HTTP calls
-(defn get+ [family-method & [{:keys [query]}]]
+(defn get+ [family-method {:keys [query token]}]
   (http/fetch-json+ (str base-uri family-method
                          (when query
                            (str "?" (uri/map->query-string query))))
                     (j/lit {:method "GET"
                             :Content-Type "application/json; charset=utf-8"
-                            :headers {:Authorization (str "Bearer " bot-token)}})))
+                            :headers {:Authorization (str "Bearer " token)}})))
 
 ;; XXX may or may not work for the rest of the API; breaks on chat.PostMessage for unknown reasons I suspect are related to node-fetch possibly mixing URL parameters with a JSON body, which makes Slack choke with "channel_not_found"
-(defn post+ [family-method body]
-  (http/fetch-json+ (str base-uri family-method)
-                    (j/lit {:method "post"
-                            :Content-Type "application/json; charset=utf-8"
-                            :headers {:Authorization (str "Bearer " bot-token)}
-                            :body (clj->js body) #_(.stringify js/JSON (clj->js body))})))
+(defn post+ [family-method {:keys [body query token]}]
+  (http/fetch-json+ (str base-uri family-method (when query
+                                                  (str "?" (uri/map->query-string query))))
+                    #js{:method "post"
+                        :Content-Type "application/json; charset=utf-8"
+                        :headers #js{:Authorization (str "Bearer " token)}
+                        :body (clj->js body) #_(.stringify js/JSON (clj->js body))}))
 
 (tasks/register-var! `post+)
 
-(defn post-query-string+ [family-method query-params]
```

Comment:
Is this no longer a problem?

Reply:
I just moved all usages to the `post+` url which accepts `:query` and `:body` params for the same purpose